testing/unittests/jlibtests.cpp

```diff
@@ -2046,6 +2049,21 @@ element1: !el scalarvalue
         CPPUNIT_ASSERT(na1->isArray(nullptr));
         CPPUNIT_ASSERT(na2->isArray(nullptr));
     }
+    void testSpecialTags()
+    {
+        //Check a null tag is supported and preserved.
+        static constexpr const char * jsonMarkup = R"!!({
+   "": "default",
+   "end": "x"
+  })!!";
+
+        Owned<IPropertyTree> json = createPTreeFromJSONString(jsonMarkup, ipt_none, ptr_ignoreWhiteSpace, nullptr);
+
+        //if we want the final compares to be less fragile (test will have to be updated if formatting changes) we could reparse and compare trees again
```

Comment:
@afishbeck I haven't checked what it gets converted to, because that might change we you implement support for a/b and other oddities.

Reply:
@ghalliday, Yes looks good.  I will expand on this a bit, and make sure both read and write work, in my expanded json name support PR.